### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-docker.git`](https://github.com/cisagov/skeleton-docker.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/postfix-docker.git postfix-docker
    cd postfix-docker
    git remote add skeleton https://github.com/cisagov/skeleton-docker.git
    git remote set-url --push skeleton no_push
    git switch develop
    git checkout -b lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the
   branch, `commit`, and `push` your changes:

    ```console
    git add README.md docker-compose.yml 
    git commit
    git push --force --set-upstream origin lineage/skeleton
    ```

    Note that you may _append_ to the default merge commit message
    that git creates for you, but *please do not delete the existing
    content*.  It provides useful information about the merge that is
    being performed.

1. Wait for all the automated tests to pass.

1. Check the "Everything is cool" checkbox below:

    - [x] ✌️ The conflicts in this pull request have been resolved.

1. Mark this draft pull request "Ready for review".

------------

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

**Review by jsf9k — CHANGES_REQUESTED** [review]
These changes look good to me, but we should wait to merge until I create the [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker) PR that uses the latest version numbers in the README.md.

**Comment by dav3r** [comment]
> These changes look good to me, but we should wait to merge until I create the [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker) PR that uses the latest version numbers in the README.md.

Agreed!  It will also allow us to ensure that this PR agrees with the finalized language that we put in the skeleton for exposed ports.

**Review by jsf9k — APPROVED** [review]
Thanks for hanging in there and waiting for that other PR to clear!

**Review by dav3r — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -15,55 +15,279 @@ Creates a Docker container with an installation of the
...
+versions](https://semver.org) of the underlying postfix project that they
+containerize.  It is recommended that most users use a version tag (e.g.
+`:0.0.4`).
 
-- `PRIMARY_DOMAIN`: the domain of the mail server
-- `RELAY_IP`: (optional) an IP address that is allowed to relay mail without authentication
+| Image:tag | Description |
+|-----------|-------------|
+|`cisagov/postfix:0.0.4`| An exact release version. |
+|`cisagov/postfix:0.4`| The most recent release matching the major and minor version numbers. |
```
Comment (mcdonnnj) [inline README.md]:
```suggestion
|`cisagov/postfix:0.0`| The most recent release matching the major and minor version numbers. |
```

File: README.md [hunk 2]
```diff
@@ -15,55 +15,279 @@ Creates a Docker container with an installation of the
...
+secrets](https://docs.docker.com/engine/swarm/secrets/).  Passing sensitive
+values like your credentials can be more secure using secrets than using
+environment variables.  See the
+[secrets](#secrets) section below for a table of all supported secret files.
+
+1. To use secrets, populate the following files in the `src/secrets` directory:
+
+- `fullchain.pem`
+- `privkey.pem`
+- `users.txt`
```
Comment (mcdonnnj) [inline README.md:95]: Unrelated to this PR, but we don't have a `.gitignore` entry to prevent something in `src/secrets` from being added accidentally. We should probably get an issue in to fix that if the README is going to suggest putting actual secrets in the directory.

File: README.md [hunk 3]
```diff
@@ -15,55 +15,279 @@ Creates a Docker container with an installation of the
...
+| Name  | Purpose | Default |
+|-------|---------|---------|
+| `RELAY_IP` | An IP address that is allowed to relay mail without authentication. | `null` |
+
+## Secrets ##
+
+| Filename     | Purpose |
+|--------------|---------|
+| `fullchain.pem` | Public key for the postfix server. |
+| `privkey.pem` | Private key for the postfix server. |
```
Comment (mcdonnnj) [inline README.md]:
```suggestion
| `fullchain.pem` | Public key for the Postfix server. |
| `privkey.pem` | Private key for the Postfix server. |
```

File: README.md [hunk 4]
```diff
@@ -15,55 +15,279 @@ Creates a Docker container with an installation of the
...
-The sample [docker composition](docker-compose.yml) publishes the
-exposed ports at 1025, 1587, and 1993.
+1. Recreate and run the container by following the [previous instructions](#running-with-docker).
 
-### Environment Variables ###
+## Image tags ##
 
-Two environment variables are used to generate the configurations at runtime:
+The images of this container are tagged with [semantic
+versions](https://semver.org) of the underlying postfix project that they
```
Comment (mcdonnnj) [inline README.md]:
```suggestion
versions](https://semver.org) of the underlying Postfix project that they
```

**Review by mcdonnnj — APPROVED** [review]
Just some small suggestions to use the capitalization the Postfix project uses and a correction for a version number.

File: README.md (hunk 2 above)
Comment (dav3r) [reply README.md:95]: > Unrelated to this PR, but we don't have a `.gitignore` entry to prevent something in `src/secrets` from being added accidentally. We should probably get an issue in to fix that if the README is going to suggest putting actual secrets in the directory.

I suppose we could rename `src/secrets` to something like `src/example_secrets` and then create a new, empty `src/secrets` and some explanatory text in the README.  Thoughts from the rest of @cisagov/team-ois?

Comment (mcdonnnj) [reply README.md:95]: This seems like a reasonable enough approach to me.

Comment (dav3r) [reply README.md:95]: @jsf9k - What sayeth thou?

Comment (jsf9k) [reply README.md:95]: This is at least the second time you guys have brought up this concern about someone accidentally committing real secrets in `src/secrets`, so we should probably look at fixing the issue in [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker).  Isn't that directory only present and non-empty in order to give the example `docker-compose.yml` file from the skeleton fake data to use when the tests are run?  If that is the case, then I'd prefer we did `git mv -r src/secrets tests && git mv docker-compose.yml tests` in [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker) and correct references in those files and others as necessary.  I think that's where that testing-specific jazz belongs.  The `README.md` can point there to highlight an example composition and secrets files.

TL;DR:  I don't think of Docker secrets as part of the source of a Docker image.

N.B.:  You can't really create an empty directory in `git`.  If there isn't any content there then that directory won't be included when you `git clone`.

Comment (dav3r) [reply README.md:95]: I created https://github.com/cisagov/skeleton-docker/issues/68 to track this issue.  @mcdonnnj - is that sufficient for now to resolve this conversation?

**Review by jsf9k — APPROVED** [review]

**Review by mcdonnnj — APPROVED** [review]
:shipit:
